pywbem/_mof_compiler.py

```diff
@@ -2834,7 +2842,9 @@ def compile_string(self, mof, ns, filename=None):
           ns (:term:`string`):
             The name of the CIM namespace in the associated CIM repository
             that is used for lookup of any dependent CIM elements, and that
-            is also the target of the compilation.
+            is also the target of the compilation. The compiler uses the
+            connection default_namespace if ns is None. If the MOF contains a
+            namespace pragma, that namespace will be used.
```

Comment:
I don't think this is correct. The namespace defined in MOF is only used from that point on in the MOF file defining it, and the previous target namespace is used upon encountering the end of that MOF file.

Reply:
Cleaned up

Reply:
It now has the same text on precedence I commented on in my other comments. Simply saying that the pragma has precedence does not sufficiently explain it, because it only precedes from the point on where it appears in the MOF, but not before. Also, we need to describe somewhere that the pragmas have a file local scope and are nested (i.e. end of file activates the namespace that was active at begin of file).